supabase/functions/send-sms/README.md

```diff
@@ -0,0 +1,171 @@
+
+
+
+## Configuration
+
+Copy this folder to the functions directory of your CLI project. In this case, my folder is located at `~/Desktop/auth_hooks/` so I will run:
+
+```bash
+cp -r ../send_sms/ ~/Desktop/auth_hooks/supabase/functions/send_sms/`
+```
+
+Copy the following environment variables from [the Twilio Console](https://console.twilio.com/) into your `.env.local` file:
+
+```bash
+TWILIO_ACCOUNT_SID=""
+TWILIO_AUTH_TOKEN=""
+TWILIO_PHONE_NUMBER=""
+```
+
+Generate an `AppHash` via [these steps set out by Google](https://developers.google.com/identity/sms-retriever/verify#computing_your_apps_hash_string) and insert the AppHash into your `.env.local`
+
+```bash
+APP_HASH="<YOUR_APP_HASH>"
+```
+
+Next, head to the `Auth > Hooks > Send SMS > HTTP > Secret > Generate Secret` page to generate a secret. The Hook Secret should start with the version identifier, `v1,` followed by the signing prefix, `whsec_`, and finally the base64 secret. For instance, if your secret is `<secret>` then  you should include the following line
+
+```bash
+SEND_SMS_HOOK_SECRET=v1,whsec_<secret>
+```
+
+in your `.env.local file`.
+
+
+## Testing The Hook
+
+Supabase follows the Standard Webhooks Specification, which is a set of guidelines used to align how webhooks are implemented. The specification describes how to secure a payload to guard against malicious attackers. Constructing a specification compliant payload can be tricky, so we use [the Simulate tool](https://www.standardwebhooks.com/simulate) provided by the specification Standard Webhooks Specification to test our Hook. In practice, you can use any HTTP function as a hook. However, for convenience and also in the spirit of dogfooding, we will describe the use of the tool with a Supabase Edge Function.
+
+
+### The Standard Webhooks Verification Tool
+
+You can use the tool with an Edge Function run locally, via the CLI, or an Edge Function deployed on the platform. We will term deployed Edge Functions as "Production" functions. 
+
+For both Local and Production Edge Functions you will need to fill the following fields:
+
+### Destination URL
+
+This is the endpoint where your Hook is running.
+
+**Local**: `https://127.0.0.1:54321/functions/v1/<your_function_name>`
+
+**Production**: `https://<project-ref>.functions.supabase.co/<function-name>`
+
+### Raw Payload
+
+Payload that Auth sends to the Hook. For the Send SMS Hook, the payload contains these entries:
+
+1. `user` - contains information describing a user, including their phone number, which we wish to send a message to.
+2. `sms` - contains metadata specific to SMS sending, including the One Time Pin.
+
+For convenience, we have attached a mock payload below. Ensure that you fill in the corresponding `phone` and `otp` entries:
+
+```
+{
+  "user": {
+    "id": "daf04a9b-faac-4425-adb0-53872dee0b06",
+    "aud": "authenticated",
+    "role": "authenticated",
+    "email": "",
```

Comment:
This will likely go out of sync, in the near future would like to have a versioned copy of the JSONSchema and reference that in the CLI to generate a payload